controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -795,8 +795,9 @@ func (pp *portPublisher) updateEndpoints(endpoints *corev1.Endpoints) {
 func (pp *portPublisher) addEndpointSlice(slice *discovery.EndpointSlice) {
 	newAddressSet := pp.endpointSliceToAddresses(slice)
 	for id, addr := range pp.addresses.Addresses {
-		newAddressSet.Addresses[id] = addr
-		newAddressSet.LocalTrafficPolicy = pp.localTrafficPolicy
+		if _, ok := newAddressSet.Addresses[id]; !ok {
+			newAddressSet.Addresses[id] = addr
```

Comment:
I notice that the assignment of `newAddressSet.LocalTrafficPolicy = pp.localTrafficPolicy` has been removed.  According to my reading of `endpointSliceToAddresses`, the `newAddressSet` will always have the traffic policy from `pp` anyway, so this assignment was redundant.  Is that correct?

Reply:
That's right